src/mpid/common/shm/mpidu_init_shm.c

```diff
@@ -0,0 +1,109 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+#include <mpidimpl.h>
+#include "mpidu_init_shm.h"
+#include "build_nodemap.h"
+
+static int local_size;
+static int my_local_rank;
+static MPIDU_shm_seg_t memory;
+static MPIDU_shm_barrier_t *barrier;
+static void *baseaddr;
+
+int MPIDU_Init_shm_seg_init(int rank, int size, int *nodemap)
+{
```

Comment:
How about remove the `_seg_` in the name? "seg" implies multiple of them, but we are only doing one and only "init_shm" block.

Reply:
In addition, I personally prefer `MPIDU_init_shm_...`, ie without capital after the namespace. We kind of been inconsistent with it, for example, we used `MPIDU_shm_seg_commit`. The first capital after capital namespace is the MPI standard convention. It might be a good practice to differentiate MPI-names from internal names with slightly different convention. What do you think?

Reply:
I will remove segment. As for the capital `Init` I agree that we are inconsistent and create confusion. We should agree upon a convention and stick to it. Otherwise it's hard to argue whether we should or not use capital letters for internal functions. Anything has same right to exist :)

Reply:
Agreed. It is up to you. I am ok to use `MPIDU_Init_shm_...` until we establish the convention.